torch/csrc/jit/tensorexpr/schedule.cpp

```diff
@@ -282,14 +284,23 @@ class Vectorizer : public IRMutator {
   const Expr* start_ = nullptr;
 };
 
-Stmt* Vectorize(const Stmt* stmt) {
-  const For* f = dynamic_cast<const For*>(stmt);
+void LoopNest::Vectorize(Stmt* stmt) {
+  For* f = dynamic_cast<For*>(stmt);
   if (!f) {
-    throw std::runtime_error("Statement is not a For loop!");
+    return;
+  }
+
+  Block* b = dynamic_cast<Block*>(f->get_parent());
+  if (!b) {
+    return;
   }
 
   Vectorizer v;
-  return v.vectorize(f);
+  try {
+    Stmt* new_f = v.vectorize(f);
```

Comment:
The original statement needs to be cloned either here or inside the vectorizer.

Reply:
Fixed